arches/settings_utils.py

```diff
@@ -1,24 +1,67 @@
-import importlib
 import json
 import os
 import site
 import sys
+from contextlib import contextmanager
 
-from pathlib import Path
+import django
+from django.apps import apps, AppConfig
+from django.conf import global_settings, settings
+
+
+@contextmanager
+def move_to_end_of_sys_path(*paths):
+    _orig_sys_path = sys.path[:]
+    for path in paths:
+        if path in sys.path:
+            sys.path.remove(path)
+            sys.path.append(path)
+    try:
+        yield
+    finally:
+        sys.path = _orig_sys_path
+
+
+def arches_applications():
+    from arches import settings as core_settings
+
+    return [
+        mod.name for mod in arches_applications_modules(core_settings.INSTALLED_APPS)
```

Comment:
nit: `module`

Reply:
Can do in this case. Is it the case you prefer multi-line blocks over slightly abbreviated names?

Reply:
Yep, every time. I'd rather have an 86-char variable name that tells you exactly what the thing is if necessary, rather an abbreviate it.

Although we're not enforcing it, I like [Google's python styleguide](https://google.github.io/styleguide/pyguide.html#316-naming)

And always [PEP20](https://peps.python.org/pep-0020/#the-zen-of-python) -- which could be argued that this falls under the 2nd aphorism 😀

Reply:
Thanks for the link, there's some great stuff in there.

Just to add to where I'm coming from: in the google style guide, under the single character section, it says "Generally speaking, descriptiveness should be proportional to the name’s scope of visibility." I think that describes why I was okay with `mod` here. It deemphasizes the comprehension variable that falls out of scope immediately. Making it take up more space gives it more importance, whereas abbreviating it suggests to the developer it’s less important than the variable being iterated over and the variable holding the result.

If I encountered a new code base where every list comprehension variable was given as much importance as the other variables, taking up multi-line blocks etc., I would find it a little harder to get up to speed.

This isn’t a list comprehension example, but earlier in that style guide, this signature is given as a blessed example, abbreviating `imag` for `imaginary`:
```py
Yes: def complex(real, imag=0.0): return Magic(r=real, i=imag)
```

...but it's a little hard to tell if that was intentional, because that example was focused on the spaces around `=`, not comprehensions.

In the comprehension section, I see:
```py
result = [mapping_expr for value in iterable if filter_expr]
```
So `mapping_expr` instead of `mapping_expression`, same with `filter_expr` etc.

***
I won't ask to change module back to mod, but I do think we need to use common sense and not rely on a one-size-fits-all rule. I also fully support the idea of erring on the side of longer names. It was a bit forbidding to work on the rdffile.py file recently, which uses a lot of one-and-two-char names like `vi`.

Reply:
No that's fair, and honestly a nitpicky on my part. Appreciate it and I'll be mindful in the future about letting some short-naming slide 👍